crystal_diffusion/oracle/lammps.py

```diff
@@ -0,0 +1,65 @@
+"""Call LAMMPS to get the forces and energy in a given configuration."""
+import os
+from pathlib import Path
+from typing import Dict, Tuple
+
+import lammps
+import numpy as np
+import pandas as pd
+import yaml
+from pymatgen.core import Element
+
+from crystal_diffusion import DATA_DIR
+
+
+def get_energy_and_forces_from_lammps(positions: np.ndarray,
+                                      box: np.ndarray,
+                                      atom_types: np.ndarray,
+                                      atom_type_map: Dict[int, str] = {1: 'Si'},
+                                      tmp_work_dir: str = './',
+                                      pair_coeff_dir: Path = DATA_DIR) -> Tuple[float, pd.DataFrame]:
+    """Call LAMMPS to compute the forces on all atoms in a configuration.
+
+    Args:
+        positions: atomic positions as a n_atom x spatial dimension array
+        box: spatial dimension x spatial dimension array representing the periodic box. Assumed to be orthogonal.
+        atom_types: n_atom array with an index representing the type of each atom
+        atom_type_map (optional): map from index representing an atom type to a description of the atom.
+            Defaults to {1: "Si"}
+        tmp_work_dir (optional): directory where the LAMMPS output will be written (and deleted). Defaults to ./
+        pair_coeff_dir (optional): directory where the potentials as .sw files are stored. Defaults to DATA_DIR
+
+    Returns:
+        energy
+        dataframe with x, y, z coordinates and fx, fy, fz information in a dataframe.
+    """
+    n_atom = positions.shape[0]
+    assert atom_types.shape == (n_atom, ), f"Atom types should match the number of atoms. Got {atom_types.shape}."
+    lmp = lammps.lammps()  # create a lammps run
+    lmp.command(f"region simbox block 0 {box[0, 0]} 0 {box[1, 1]} 0 {box[2, 2]}")  # TODO what if box is not orthogonal
```

Comment:
Here I suggest putting a protective `assert` to prevent the user from making a silent mistake. Something like

`assert np.allclose(box, np.diag(np.diag(box))), "only diagonal 'box' are valid"`